File: Angular/src/app/Models/CarDTO.ts
import { Image } from "./image";

export class CarDTO {
  id: number;
  name: string;
  model: string;
  nb_doors: number;
  nb_places: number;
  address: string;
  description: string;
  price_per_day: number;
  registration_num: string;
  gearbox: string;
  photo?: File;
  imageData?: string;
  fileType?: string;

  constructor(car: any) {
    this.id = car.id;
    this.name = car.name;
    this.model = car.model;
    this.nb_doors = car.nb_doors;
    this.nb_places = car.nb_places;
    this.address = car.address;
    this.description = car.description;
    this.price_per_day = car.price_per_day;
    this.registration_num = car.registration_num;
    this.gearbox = car.gearbox;
    this.photo = car.photo;
    this.imageData = car.imageData;  
    this.fileType = car.fileType;

  }
}

File: SpringBoot/src/main/java/com/mycompany/rentCar/CarDTO/CarDTO.java
package com.mycompany.rentCar.CarDTO;

import com.mycompany.rentCar.Entities.Cars;
import com.mycompany.rentCar.Entities.Image;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CarDTO {
    private Long id;
    private String name;
    private String model;
    private int nb_doors;
    private int nb_places;
    private String address;
    private String description;
    private float price_per_day;
    private String registration_num;
    private String gearbox;
    private long imageId;
    private String imageName;
    private String imageFileType;

    private byte[] imageData;
    public CarDTO(Cars car) {
        this.id = car.getId();
        this.name = car.getName();
        this.model = car.getModel();
        this.nb_doors = car.getNb_doors();
        this.nb_places = car.getNb_places();
        this.address = car.getAddress();
        this.description = car.getDescription();
        this.price_per_day = car.getPrice_per_day();
        this.registration_num = car.getRegistration_num();
        this.gearbox = car.getGearbox();

        Image image = car.getImage();
        if (image != null) {
            this.imageId = image.getId();
            this.imageName = image.getFileName();
            this.imageFileType = image.getFileType();
            this.imageData= image.getData();
        }
    }
}
